### PR TITLE
[export] update dynamic shapes section

### PR DESCRIPTION
**Pull request by pianpwk**
## Description
Updates the export dynamic shapes tutorial to be more up-to-date with our new APIs.

## Checklist

- [ ] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [ ] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

cc @avikchaudhuri @gmagogsfm @zhxchen17 @tugsbayasgalan @angelayi @suo @ydwu4

### CODE REVIEW

File: intermediate_source/torch_export_tutorial.py [hunk 1]
```diff
@@ -304,237 +304,315 @@ def false_fn(x):
...
+    "x": (Dim.AUTO,),
+    "y": (Dim.AUTO, Dim.AUTO),
+    "z": (Dim.AUTO,),
+}
+ep = export(model, (w, x, y, z), dynamic_shapes=dynamic_shapes)
 
-    def forward(self, x, y):
-        return torch.nn.functional.relu(self.lin(x + y), inplace=True)
+######################################################################
+# Before we look at the program that's produced, let's understand what specifying ``dynamic_shapes`` entails,
```
Comment (yushangdi) [inline intermediate_source/torch_export_tutorial.py:358]: nit: can we make this the regular text instead of a comment? It's a little hard to read because of the color and line breaks.

Comment (yushangdi) [reply intermediate_source/torch_export_tutorial.py:358]: similarly for the other chunks of long texts below

Comment (pianpwk) [reply intermediate_source/torch_export_tutorial.py:358]: Oh hmm I thought this was fixed, my bad

Comment (pianpwk) [reply intermediate_source/torch_export_tutorial.py:358]: Actually I think they just need to be regenerated.. not sure how

Comment (yushangdi) [reply intermediate_source/torch_export_tutorial.py:358]: > Actually I think they just need to be regenerated.. not sure how

I thought the preview here should be the regenerated one? They are still comments here. https://docs-preview.pytorch.org/pytorch/tutorials/3183/intermediate/torch_export_tutorial.html#constraints-dynamic-shapes
